akka-stream-tests/src/test/scala/akka/stream/scaladsl/QueueSourceSpec.scala

```diff
@@ -258,13 +258,16 @@ class QueueSourceSpec extends StreamSpec {
 
     "fail offer future when stream is completed" in assertAllStagesStopped {
       val s = TestSubscriber.manualProbe[Int]()
-      val queue = Source.queue(1, OverflowStrategy.dropNew).to(Sink.fromSubscriber(s)).run()
-      val sub = s.expectSubscription()
+      val queue = Source
+        .queue[Int](1, OverflowStrategy.dropNew)
+        .merge(Source.empty[Int], eagerComplete = true)
```

Comment:
This is a pretty intricate way of introducing the cancellation... does the `sub.cancel()` no longer work?

Reply:
It works, although it returned a different stream "failure". I'm a little confused about this TBH since I'm not sure why a failure should be returned if a completion occurs.

Reply:
> I'm a little confused about this TBH since I'm not sure why a failure should be returned if a completion occurs.

That's a good point: if downstream is no longer interested in more elements, should that look like a 'failure' to upstream or should those elements just disappear into the void? I can see pros and cons for either interpretation...